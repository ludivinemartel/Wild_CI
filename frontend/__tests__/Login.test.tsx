import Login from "@/pages/auth/login";
import { fireEvent, render, waitFor } from "@testing-library/react";
import { LOGIN } from "@/requetes/queries/auth.queries";
import { MemoryRouterProvider } from "next-router-mock/MemoryRouterProvider";
import { MockedProvider } from "@apollo/client/testing";

jest.mock("next/router", () => ({
  ...jest.requireActual("next-router-mock"),
  push: jest.fn(),
}));

const mocks = [
  {
    request: {
      query: LOGIN,
      variables: {
        infos: { email: "test@mondomaine.com", password: "motdepasse" },
      },
    },
    result: {
      data: {
        login: {
          success: true,
          message: "Bienvenue!",
        },
      },
    },
  },
];

it("Snapshot du composant login", async () => {
  const { container, getByText, getByPlaceholderText  } = render(
    <MemoryRouterProvider url="/auth/login">
      <MockedProvider mocks={mocks} addTypename={false}>
        <Login />
      </MockedProvider>
    </MemoryRouterProvider>
  );

  // Remplissage du formulaire avec des données de test
  fireEvent.change(getByPlaceholderText("Indiquez votre email"), {
    target: { value: "test@mondomaine.com" },
  });
  fireEvent.change(getByPlaceholderText("Indiquez votre mot de passe"), {
    target: { value: "motdepasse" },
  });

  // Soumission du formulaire
  fireEvent.submit(getByText("Connexion"));

  // On attend un certain temps pour permettre à onCompleted de s'exécuter
  await waitFor(() => {}, { timeout: 1000 });

  // On vérifie le snapshot après que onCompleted a eu le temps de s'exécuter
  await waitFor(
    () => {
      expect(container).toMatchSnapshot();
    },
    { timeout: 10000 }
  );
});
