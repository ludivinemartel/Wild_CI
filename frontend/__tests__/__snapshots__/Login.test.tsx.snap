// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot du composant login 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center justify-between p-24"
  >
    <form>
      <h1
        class="font-bold text-lg mb-8"
      >
        Connexion
      </h1>
      <div>
        <input
          name="email"
          placeholder="Indiquez votre email"
          type="text"
        />
      </div>
      <div>
        <input
          name="password"
          placeholder="Indiquez votre mot de passe"
          type="password"
        />
      </div>
      <input
        type="submit"
      />
      <div>
        <span
          class="text-red-500"
        />
        <span
          class="text-blue-500"
        >
          Bienvenue!
        </span>
      </div>
      <a
        href="/auth/reset"
      >
        mot de passe oubli√©?
      </a>
    </form>
  </main>
</div>
`;
